<launch>

  <arg name="position" default="false"/>  
  <arg name="torque" default="true"/>
  <arg name="torque_extension" default="false"/>
  <arg name="verbose" default="true"/>


  <!-- Load the URDF into the ROS Parameter Server
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find rrrbot_files)/urdf/rrrbot.xacro'" />
 -->
  <!-- Show in Rviz   -->
  <!--node name="rviz" pkg="rviz" type="rviz" args="-d $(find rrrbot_files)/rviz/rrrbot.rviz"/-->


  <!-- Include gazebo launch file rrbot_world.launch -->
  <include file="$(find gmm_obs)/launch/robot_world.launch">
    <arg name="verbose" value="$(arg verbose)"/>
  </include>


  <!-- Run joint position controllers if true -->
  <!--group if="$(arg position)"-->
   
    <!-- Include ros_control launch file to load joint position controllers -->
    <!--include file="$(find rrrbot_files)/launch/rrrbot_control.launch" /-->

    <!-- Create node to control joint positions using joint controller -->
    <!---node name="rrrbot_joint_positions_node" pkg="rrrbot_files" type="rrrbot_position_controller.py" output="screen" respawn="true" />
  </group-->


  <!-- Run joint torque controllers if true -->
  <group if="$(arg torque)">
    <!-- Include ros_control launch file to load joint torque controllers -->
    <include file="$(find gmm_obs)/launch/torque_control.launch" />

    <!-- Create node to control joint torques using effort controller -->
    <node name="pendulum_joint_torque_node" pkg="gmm_obs" type="torque_controller.py" output="screen" respawn="true" />
  </group>


  <!-- Run joint torque controllers for added mass at end if true -->
  <!--group if="$(arg torque_extension)"-->
    <!-- Include ros_control launch file to load joint torque controllers -->
    <!--include file="$(find rrrbot_files)/launch/rrrbot_torque_control.launch" /-->

    <!-- Create node to control joint torques using effort controller -->
    <!--node name="pendulum_joint_torque_node" pkg="rrrbot_files" type="rrrbot_torques_controller_extension.py" output="screen" respawn="true" />
  </group-->


</launch>
